bin/update-peer-deps.js

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env node
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: loopback-next
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * This is an internal script to update peer dependencies based on released
+ * LoopBack package versions.
+ */
+'use strict';
+
+const fs = require('fs');
+const semver = require('semver');
+
+const Project = require('@lerna/project');
+
+async function updatePeerDeps() {
+  // List all packages within the monorepo
+  const project = new Project(process.cwd());
+  const packages = await project.getPackages();
+
+  // Build a map of module name/version pairs
+  const lbModuleVersions = {};
+  for (const p of packages) {
+    lbModuleVersions[p.name] = p.version;
+  }
+
+  for (const p of packages) {
+    // Load package.json
+    const pkgJsonFile = p.manifestLocation;
+    const pkgJson = require(pkgJsonFile);
+    let updated = false;
+    if (pkgJson.peerDependencies) {
+      // Check all entries in peerDependencies
+      for (const d in pkgJson.peerDependencies) {
+        const pkgVersion = lbModuleVersions[d];
+        if (pkgVersion) {
+          // Update the version range to be `^<major>.0.0`
+          pkgJson.peerDependencies[d] = `^${semver.major(pkgVersion)}.0.0`;
```

Comment:
Should we use the actual `pkgVersion`, since that's the version we are testing in our CI? If we use `MAJOR.0.0`, then it's possible to create a situation where the peer dep spec is too wide, because the extension depends on features introduced in semver-minor version after `MAJOR.0.0`.

Another aspect to consider: removing a version from `peerDependencies` range is a breaking change, similarly to how dropping support for a Node.js major version is a breaking change too. 

It would be great if `lerna` was handling these complexities automatically for us, see the discussion in https://github.com/lerna/lerna/issues/955 and https://github.com/lerna/lerna/issues/1018

Reply:
Let's be conservative to use the `^<pkgVersion>` for peer dependencies as a release is cut.